pkg/dag/dag.go

```diff
@@ -0,0 +1,305 @@
+package dag
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/kendru/darwin/go/depgraph"
+)
+
+// Graph represents a directed graph.
+type Graph struct {
```

Comment:
@frelon Out of curiosity, where is all this `dag` code coming from? I can´t see how it relates to the deps cleanup.

Reply:
It's coming from `github.com/spectrocloud-labs/herd`, but since that is only a thin wrapper around `github.com/kendru/darwin/go/depgraph` that pulls in other deps like `github.com/samber/lo`, I copied it here with some small changes so we could drop `github.com/samber/lo`.